pkg/promotion/env_location.go

```diff
@@ -0,0 +1,43 @@
+package promotion
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+	"strings"
+
+	"github.com/rhd-gitops-example/services/pkg/util"
+)
+
+type EnvLocation struct {
+	RepoPath string // URL or local path
+	Branch   string
+	Folder   string
+}
+
+func (env EnvLocation) IsLocal() bool {
+	parsed, err := url.Parse(env.RepoPath)
+	if err != nil {
+		log.Printf("Error while parsing URL for environment path: '%v', assuming it is local.\n", err)
```

Comment:
Shouldn't this be an errorf and without the \n? Is it safe to print this (no token leakage)?

Reply:
I don't think there is a token leakage. `env.RepoPath` is just the URL, the token/username are added later by `addCredentialsIfNecessary` in the `cloneRepo` function.

It can return an error, that's probably better.